### PR TITLE
[MODULAR] Fix Duplicated Ghostrole Outfit Skillchips and Backpack Contents!

### PR DESCRIPTION
**Pull request by RimiNosha**
## About The Pull Request
Fixes a bug where skillchips and backpack contents are applied twice, due to `human/proc/equipOutfit` being called twice if prefs are loaded!

Technically fully modularizes ghostrole prefs, though it didn't really make sense to me to move to proc to a modular file.

Oh- and also adds an extra argument to `equip_outfit_and_loadout` that allows ghostrole outfit overrides to be passed!

There is a very minor issue at the moment where you're given control of the mob spawned before the prompt goes away, so if anyone can propose a good solution, that'd be very helpful!

## How This Contributes To The Skyrat Roleplay Experience
Duplicated items fill out backpacks with useless clutter! Also balance!

## Changelog

:cl:
fix: Fixed prefloaded ghostroles duplicating their skillchips and backpack contents.
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/loadouts/loadout_ui/loadout_outfit_helpers.dm [hunk 1]
```diff
@@ -30,6 +30,13 @@
 	else
 		CRASH("Outfit passed to equip_outfit_and_loadout was neither a path nor an instantiated type!")
 
+	if(outfit_override)
+		for(var/outfit_var in outfit_override)
+			if(!ispath(outfit_override[outfit_var]) && !isnull(outfit_override[outfit_var]))
+				CRASH("outfit_override var on [real_name] spawner has incorrect values! it must be an assoc list with outfit \"var\" = path | null")
+			outfit.vars[outfit_var] = outfit_override[outfit_var]
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/loadouts/loadout_ui/loadout_outfit_helpers.dm:37]: What's the purpose of this?

Also directly accessing vars isn't great in the slightest.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/loadouts/loadout_ui/loadout_outfit_helpers.dm (hunk 1 above)
Comment (RimiNosha) [reply modular_skyrat/modules/loadouts/loadout_ui/loadout_outfit_helpers.dm:37]: It's to support whatever admin overrides are set for custom spawners, or at least that's what the TG comments say.

Comment (RimiNosha) [reply modular_skyrat/modules/loadouts/loadout_ui/loadout_outfit_helpers.dm:37]: We've never really had issues with it before from what I know, so I'm fine with reverting that change if you want me to.

Comment (GoldenAlpharex) [reply modular_skyrat/modules/loadouts/loadout_ui/loadout_outfit_helpers.dm:37]: > or at least that's what the TG comments say.

Where did it say that?

**Comment by Gandalf2k15** [comment]
Fix up conflicts and address comments then remake or reopen

**Pull request closed without merging**